app/assets/javascripts/components/articles/article.jsx

```diff
@@ -44,6 +44,8 @@ const Article = createReactClass({
   render() {
     const ratingClass = `rating ${this.props.article.rating}`;
     const ratingMobileClass = `${ratingClass} tablet-only`;
+    const isDeleted = this.props.article.deleted;
+    const deletedMessage = I18n.t('articles.deleted_message');
```

Comment:
Since this never varies and is only rarely used, it might make sense to implement this as a constant outside of the class. That way, it will only call I18n.t once, instead of once for every time an Article renders.

I doubt this one thing makes any practical performance difference, but I do think that overall, rendering large numbers of Article components can cause real performance problems.

It also just streamlines the logic slightly, as generating the `deletedMessage` from i18n is not really part of the core task of rendering this component.